File: .gemini/antigravity/scratch/CLI-agent-claude-cli-agent-planning-Z804h/python/aurex/config/paths.py
"""Platform-aware paths for caching and configuration."""

import platform
import os
from pathlib import Path


def get_cache_dir(app_name: str = "aurex") -> Path:
    """Get the platform-specific cache directory."""
    system = platform.system()
    home = Path.home()
    
    if system == "Windows":
        base_dir = Path(os.environ.get("APPDATA", home / "AppData" / "Roaming"))
    elif system == "Darwin":
        base_dir = home / "Library" / "Caches"
    else:  # Linux
        base_dir = Path(os.environ.get("XDG_CACHE_HOME", home / ".cache"))
        
    cache_dir = base_dir / app_name
    
    try:
        cache_dir.mkdir(parents=True, exist_ok=True)
    except Exception:
        # Fallback to local directory if permissions fail
        cache_dir = Path.cwd() / f".{app_name}"
        cache_dir.mkdir(parents=True, exist_ok=True)
        
    return cache_dir


def get_config_dir(app_name: str = "aurex") -> Path:
    """Get the platform-specific config directory."""
    system = platform.system()
    home = Path.home()
    
    if system == "Windows":
        base_dir = Path(os.environ.get("APPDATA", home / "AppData" / "Roaming"))
    elif system == "Darwin":
        base_dir = home / "Library" / "Application Support"
    else:  # Linux
        base_dir = Path(os.environ.get("XDG_CONFIG_HOME", home / ".config"))
        
    config_dir = base_dir / app_name
    
    try:
        config_dir.mkdir(parents=True, exist_ok=True)
    except Exception:
        # Fallback
        return Path.cwd()
        
    return config_dir

File: .gemini/antigravity/scratch/CLI-agent-claude-cli-agent-planning-Z804h/src/config/paths.ts
import * as os from "os";
import * as path from "path";
import * as fs from "fs";

export function getCacheDir(appName = "aurex"): string {
    const platform = os.platform();
    const home = os.homedir();
    let baseDir: string;

    if (platform === "win32") {
        baseDir = process.env.APPDATA || path.join(home, "AppData", "Roaming");
    } else if (platform === "darwin") {
        baseDir = path.join(home, "Library", "Caches");
    } else {
        // Linux and others
        baseDir = process.env.XDG_CACHE_HOME || path.join(home, ".cache");
    }

    const dir = path.join(baseDir, appName);

    try {
        if (!fs.existsSync(dir)) {
            fs.mkdirSync(dir, { recursive: true });
        }
    } catch (err) {
        // If we can't create the directory, fallback to a local .aurex folder or temp
        try {
            const localDir = path.join(process.cwd(), `.${appName}`);
            if (!fs.existsSync(localDir)) {
                fs.mkdirSync(localDir, { recursive: true });
            }
            return localDir;
        } catch {
            return os.tmpdir();
        }
    }

    return dir;
}

export function getConfigDir(appName = "aurex"): string {
    const platform = os.platform();
    const home = os.homedir();
    let baseDir: string;

    if (platform === "win32") {
        baseDir = process.env.APPDATA || path.join(home, "AppData", "Roaming");
    } else if (platform === "darwin") {
        baseDir = path.join(home, "Library", "Application Support");
    } else {
        // Linux and others
        baseDir = process.env.XDG_CONFIG_HOME || path.join(home, ".config");
    }

    const dir = path.join(baseDir, appName);
    try {
        if (!fs.existsSync(dir)) {
            fs.mkdirSync(dir, { recursive: true });
        }
    } catch {
        // Fallback
        return process.cwd();
    }

    return dir;
}
